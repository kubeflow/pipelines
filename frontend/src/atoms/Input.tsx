/*
 * Copyright 2018 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 * https://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import * as React from 'react';
import TextField, { OutlinedTextFieldProps } from '@material-ui/core/TextField';
import { commonCss } from '../Css';

interface InputProps extends OutlinedTextFieldProps {
  height?: number | string;
  maxWidth?: number | string;
  width?: number;
}

const Input = (props: InputProps) => {
  const { height, maxWidth, variant, width, ...rest } = props;
  return (
    <TextField
      variant={variant}
      className={commonCss.textField}
      spellCheck={false}
      style={{
        height: !!props.multiline ? 'auto' : height || 40,
        maxWidth: maxWidth || 600,
        width: width || '100%',
      }}
      {...rest}
    >
      {props.children}
    </TextField>
  );
};

export default Input;
