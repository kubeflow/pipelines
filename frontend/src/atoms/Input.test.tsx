/*
 * Copyright 2018 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 * https://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import * as React from 'react';

import Input from './Input';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

describe('Input', () => {
  const handleChange = jest.fn();
  const value = 'some input value';

  it('renders with the right styles by default', () => {
    const tree = shallow(
      <Input onChange={handleChange('fieldname')} value={value} variant='outlined' />,
    );
    expect(toJson(tree)).toMatchSnapshot();
  });

  it('accepts height and width as prop overrides', () => {
    const tree = shallow(
      <Input
        height={123}
        width={456}
        onChange={handleChange('fieldname')}
        value={value}
        variant='outlined'
      />,
    );
    expect(toJson(tree)).toMatchSnapshot();
  });
});
