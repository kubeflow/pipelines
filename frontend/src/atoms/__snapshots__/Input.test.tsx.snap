// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input accepts height and width as prop overrides 1`] = `
<TextField
  className="textField"
  required={false}
  select={false}
  spellCheck={false}
  style={
    Object {
      "height": 123,
      "maxWidth": 600,
      "width": 456,
    }
  }
  value="some input value"
  variant="outlined"
/>
`;

exports[`Input renders with the right styles by default 1`] = `
<TextField
  className="textField"
  required={false}
  select={false}
  spellCheck={false}
  style={
    Object {
      "height": 40,
      "maxWidth": 600,
      "width": "100%",
    }
  }
  value="some input value"
  variant="outlined"
/>
`;
